code/modules/power/singularity/singularity.dm

```diff
@@ -355,9 +355,15 @@
 		if(M.stat == CONSCIOUS)
 			if (istype(M,/mob/living/carbon/human))
 				var/mob/living/carbon/human/H = M
-				if(istype(H.glasses,/obj/item/clothing/glasses/meson))
-					H << "<span class='notice'>You look directly into the [src.name], good thing you had your protective eyewear on!</span>"
-					return
+				if(istype(H.glasses,/obj/item/clothing/glasses/meson/engine))
+					var/obj/item/clothing/glasses/meson/engine/G = H.glasses
+					if(!G.mode)
+						H << "<span class='notice'>You look directly into the [src.name], good thing you had your protective eyewear on!</span>"
+						return
+				else
+					if(istype(H.glasses,/obj/item/clothing/glasses/meson))
+						H << "<span class='notice'>You look directly into the [src.name], good thing you had your protective eyewear on!</span>"
+						return

```

Comment:
you can reduce the duplicate code by using 
`if(istype(H.glasses, /obj/item/clothing/glasses/meson))`
`____var/obj/item/clothing/glasses/meson/MS = H.glasses`
`____if(MS.vision_flags == SEE_TURFS)`
`________H << "<span class='notice'>You look directly into ....`
`________return`